static/src/js/components/ContactInfo/__tests__/ContactInfo.test.js

```diff
@@ -31,6 +33,59 @@ describe('ContactInfo component', () => {
     expect(enzymeWrapper.exists()).toBeTruthy()
   })
 
+  test('should render contact info form correctly before preferences are retrieved', () => {
+    const { enzymeWrapper } = setup({
+      contactInfo: {
+        ursProfile: { mock: 'urs' }
+      }
+    })
+
+    const spinner = enzymeWrapper.find(Spinner)
+    expect(spinner.prop('size')).toEqual('x-tiny')
+    expect(spinner.prop('inline')).toEqual(true)
+    expect(spinner.length).toEqual(1)
+
+    expect(enzymeWrapper.state()).toEqual({ notificationLevel: 'VERBOSE' })
+  })
+
+  test('should render user notification level after preferences are retrieved', () => {
+    const { enzymeWrapper } = setup()
+    expect(enzymeWrapper.exists()).toBeTruthy()
+
+    const spinner = enzymeWrapper.find(Spinner)
+    expect(spinner.length).toEqual(0)
+
+    expect(enzymeWrapper.state()).toEqual({ notificationLevel: 'VERBOSE' })
+
+    enzymeWrapper.setProps({
+      contactInfo: {
+        cmrPreferences: { notificationLevel: 'INFO' },
+        ursProfile: { mock: 'urs' }
+      }
+    })
+
+    expect(enzymeWrapper.state()).toEqual({ notificationLevel: 'INFO' })
```

Comment:
This test should also assert that the spinner is no longer there, and the select field is visible after the `setProps` call